mongo/mongo.go

```diff
@@ -27,3 +35,42 @@ func (m *Mongo) Init() (session *mgo.Session, err error) {
 	session.SetMode(mgo.Strong, true)
 	return session, nil
 }
+
+// GetRecipes retrieves all recipe documents from Mongo
+func (m *Mongo) GetRecipes() ([]recipe.Response, error) {
+	s := m.Session.Copy()
+	defer s.Close()
+
+	iter := s.DB(m.Database).C(m.Collection).Find(nil).Iter()
+	defer func() {
+		err := iter.Close()
+		if err != nil {
+			log.Event(context.Background(), "error closing iterator", log.ERROR, log.Error(err))
+		}
+	}()
+
+	results := []recipe.Response{}
+	if err := iter.All(&results); err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, errs.ErrRecipeNotFound
+		}
+		return nil, err
+	}
+	return results, nil
+}
+
+// GetRecipe retrieves a recipe document
+func (m *Mongo) GetRecipe(id string) (*recipe.Response, error) {
+	s := m.Session.Copy()
+	defer s.Close()
+	var recipe recipe.Response
+	err := s.DB(m.Database).C(m.Collection).Find(bson.M{"_id": id}).One(&recipe)
+	if err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, errs.ErrRecipeNotFound
```

Comment:
Rectified in commit "Tidy code and comments"

Reply:
But now you are forcing it to always be 404 not found this is also incorrect as it could be an internal server error. You need to use the returned error to decide what the response status code should be.

If err is equal to errs.ErrRecipeNotFound then the response status code is 404; if it is any other error return 500 internal server error